### PR TITLE
Feat: added tour guide for user

### LINKED ISSUE #203: Add a tour guide
Steps for Tour guide:

1. Tour will show the landing page with Welcome modal open
    a. Click Next to entertain user with all 3 pages
    b. Click Explore OpenMetadata  to close modal and show landing page 
2. Step 2 will show the landing page route
3. From here we will prompt “Click Explore to browse and search for datasets”
4. once they click Explore,  Prompt users to type one of the sample data sets to search, “Users”
5. Search should show the dataset, and prompt users to click
6. On click, will show the dataset page and at this point we can display message,  “Understand the schema of the table and add description, Claim ownership. Add tags etc..”
7. After say 10 seconds, we can prompt saying “Tour is complete , Let's Explore ” and “End tour” button.  on clicking this should go to main landing page.

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

Added Tour guide for user
Closes #203 #728 

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :
image
image
image
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Frontend: @shahsank3t, @darth-coder00, @Sachin-chaurasiya

### CODE REVIEW

**Comment by harshach** [comment]
@ShaileshParmar11 this looks great!! awesome job
but our front page is changed :) 
cc @shahsank3t @Sachin-chaurasiya

**Comment by shahsank3t** [comment]
@harshach Yes, that's the reason this PR is in draft. Once new changes are properly implemented, Shailesh can continue on this again on the new one.

**Comment by ShaileshParmar11** [comment]
> @ShaileshParmar11 this looks great!! awesome job but our front page is changed :) cc @shahsank3t @Sachin-chaurasiya

Thank you..! and yes as soon as the new design is implemented ill make the necessary changes for this feature.

File: openmetadata-ui/src/main/resources/ui/package.json [hunk 1]
```diff
@@ -89,7 +89,8 @@
     "stream-http": "^3.2.0",
     "styled-components": "^5.2.3",
     "tailwindcss": "^2.1.4",
-    "to-arraybuffer": "^1.0.1"
+    "to-arraybuffer": "^1.0.1",
+    "@deuex-solutions/react-tutorial": "1.0.0"
```
Comment (darth-coder00) [inline openmetadata-ui/src/main/resources/ui/package.json:93]: Move it up with redoc package

File: openmetadata-ui/src/main/resources/ui/src/components/Explore/Explore.component.tsx [hunk 2]
```diff
@@ -560,7 +561,9 @@ const Explore: React.FC<ExploreProps> = ({
             showResultCount
             currentPage={currentPage}
             data={data}
-            isLoading={isEntityLoading}
+            isLoading={
+              location.pathname.includes(ROUTES.TOUR) ? false : isEntityLoading
```
Comment (darth-coder00) [inline openmetadata-ui/src/main/resources/ui/src/components/Explore/Explore.component.tsx:565]:
```suggestion
              !location.pathname.includes(ROUTES.TOUR) && isEntityLoading
```

File: openmetadata-ui/src/main/resources/ui/src/components/app-bar/Appbar.tsx [hunk 3]
```diff
@@ -235,6 +250,13 @@ const Appbar: React.FC = (): JSX.Element => {
                   const target = e.target as HTMLInputElement;
                   if (e.key === 'Enter') {
                     setIsOpen(false);
+                    // below code is for tour feature
+                    if (location.pathname.includes(ROUTES.TOUR)) {
+                      appState.currentTourPage = 'explorePage';
```
Comment (darth-coder00) [inline openmetadata-ui/src/main/resources/ui/src/components/app-bar/Appbar.tsx:255]: Should have `enum` for Tour pages

File: openmetadata-ui/src/main/resources/ui/src/components/MyDataHeader/MyDataHeader.component.tsx [hunk 4]
```diff
@@ -36,7 +36,7 @@ const MyDataHeader: FunctionComponent<Props> = ({
   entityCounts,
   ingestionCount,
 }: Props) => {
-  const { users, userTeams } = AppState;
+  const { users, userTeams, isTourOpen } = AppState;
```
Comment (darth-coder00) [inline openmetadata-ui/src/main/resources/ui/src/components/MyDataHeader/MyDataHeader.component.tsx:39]: Discard changes for this file

File: openmetadata-ui/src/main/resources/ui/src/components/common/table-data-card/TableDataCard.tsx [hunk 5]
```diff
@@ -93,10 +99,26 @@ const TableDataCard: FunctionComponent<Props> = ({
     return [...new Set(assetTags)];
   };
 
+  const handleCardClick = () => {
+    if (location.pathname.includes(ROUTES.TOUR)) {
+      AppState.currentTourPage = 'datasetPage';
```
Comment (darth-coder00) [inline openmetadata-ui/src/main/resources/ui/src/components/common/table-data-card/TableDataCard.tsx:104]: Same here

File: openmetadata-ui/src/main/resources/ui/src/react-app-env.d.ts [hunk 6]
```diff
@@ -27,3 +27,4 @@ declare module 'react-table';
 declare module 'recharts';
 declare module 'react-tutorial';
```
Comment (darth-coder00) [inline openmetadata-ui/src/main/resources/ui/src/react-app-env.d.ts:28]: Remove this now

**Review by darth-coder00 — CHANGES_REQUESTED** [review]
Added few comments

File: openmetadata-ui/src/main/resources/ui/package.json (hunk 1 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/package.json:93]: Done

File: openmetadata-ui/src/main/resources/ui/src/components/Explore/Explore.component.tsx (hunk 2 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/Explore/Explore.component.tsx:565]: updated

File: openmetadata-ui/src/main/resources/ui/src/components/app-bar/Appbar.tsx (hunk 3 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/app-bar/Appbar.tsx:255]: updated

File: openmetadata-ui/src/main/resources/ui/src/components/MyDataHeader/MyDataHeader.component.tsx (hunk 4 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/MyDataHeader/MyDataHeader.component.tsx:39]: updated

File: openmetadata-ui/src/main/resources/ui/src/components/common/table-data-card/TableDataCard.tsx (hunk 5 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/common/table-data-card/TableDataCard.tsx:104]: updated

**Review by darth-coder00 — APPROVED** [review]
LGTM!

**Pull request merged**